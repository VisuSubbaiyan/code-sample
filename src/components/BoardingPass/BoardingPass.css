
.card.boarding-pass {
  display: block;
  padding: 5px;
}
.card .flight-details, .card .boarding-details, .card .gate-details {
  display: flex;
  justify-content: space-between;
  padding: 5px;
  align-items: center;
}

.card .boarding-details {
  background: #fff;
}

.card .ontime {
  border-radius: 3px;
  padding: 3px;
  color: #eeeeee;
  background: green;
}
