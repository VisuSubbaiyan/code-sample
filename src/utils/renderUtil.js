import React from 'react';

import {Airport, BoardingPass, Ground} from '../components';

export const renderItems = items => items.map(item => {
  const { type, segment_id } = item;

  switch (type) {
    case 'GROUND': {
      return <Ground {...item} key={segment_id} className="segment-active no-top-padding" />;
    }
    
    case 'AIRPORT': {
      return <Airport {...item} key={segment_id} />;
    }

    default:
      return <BoardingPass {...item} key={segment_id} />
  }
});
