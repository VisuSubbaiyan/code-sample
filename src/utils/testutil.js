import { mount, shallow } from 'enzyme';
import { prettyPrint } from 'html';

export const mockFunction = jest.fn(value => value);

export const sampleFriend = {
  name: 'Sample friend',
  gender: 'male'
};

export const sampleId = 1;

export const createMountedComponent = ({ children, isShallow }) => {
  const mountedComponent = isShallow ? shallow(children) : mount(children);
  mountedComponent.htmlRender = () => prettyPrint(mountedComponent.html(), { indent_size: 2 });

  return mountedComponent;
};

