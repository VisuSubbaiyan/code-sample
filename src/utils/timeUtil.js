import moment from 'moment';

export const formatTime = time => {
  const tempTime = moment.duration(time);
  
  return `${tempTime.hours()}: ${tempTime.minutes()}`;
}