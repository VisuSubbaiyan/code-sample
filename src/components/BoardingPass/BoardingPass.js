import React from 'react';
// import PropTypes from 'prop-types';
import IconButton from '../IconButton';
import TimeDisplay from '../TimeDisplay';

import './BoardingPass.css';

const BoardingPass = props => (
  <div className="segment" >
    <TimeDisplay time={props.departure} />
    <div className="card boarding-pass">
      <div className="flight-details">
        <div>{props.detail.flight_number}</div>
        <div className="boarding-time">
          <TimeDisplay time={props.departure} />
        </div>
      </div>
      <div className="boarding-details">
        <h2>{props.origin.value}</h2>
        <IconButton icon="plane" />
        <h2>{props.destination.value}</h2>
      </div>
      <div className="gate-details">
        <span>{`Gate ${props.detail.gate}`}</span>
        <span>{`Seat ${props.detail.seat}`}</span>
        <span className="ontime">{'OnTime'}</span>
      </div>
    </div>
  </div>
);

export default BoardingPass;
