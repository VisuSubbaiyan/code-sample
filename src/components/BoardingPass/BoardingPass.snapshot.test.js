import React from 'react';

import { createMountedComponent } from '../../utils/testutil';

import { segments } from '../../data.json';
import BoardingPass from './BoardingPass';

describe('<BoardingPass /> should', () => {
  const mountedComponent = createMountedComponent({
    children: <BoardingPass {...segments[2]} />
  });

  it('render correctly according to props', () => {
    expect(mountedComponent.htmlRender()).toMatchSnapshot();
  })
});