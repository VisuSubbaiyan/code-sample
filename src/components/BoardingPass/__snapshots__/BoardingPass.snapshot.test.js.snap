// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BoardingPass /> should render correctly according to props 1`] = `
"<div class=\\"segment\\">
  <h3 class=\\"time\\">15: 13</h3>
  <div class=\\"card boarding-pass\\">
    <div class=\\"flight-details\\">
      <div>LH2031</div>
      <div class=\\"boarding-time\\">
        <h3 class=\\"time\\">15: 13</h3>
      </div>
    </div>
    <div class=\\"boarding-details\\">
      <h2>TXL</h2>
      <div><i class=\\"fa fa-plane\\"></i>
      </div>
      <h2>MUC</h2>
    </div>
    <div class=\\"gate-details\\"><span>Gate C12</span><span>Seat 12C</span><span class=\\"ontime\\">OnTime</span>
    </div>
  </div>
</div>"
`;
